### DISCUSSION TITLE
Deployment workflow - question from a starter

### DISCUSSION THREAD

**Discussion by FoxDawg** [statamic/cms#6890] [Q&A]
Hi guys,
being a .NET developer trying to learn new stuff, I still have to get my head around some questions concerning statamic deployment and the tools involved. I'd be glad if someone could share their opinions with me!

As for the deployment, I imagine the following workflow example:
1. Create site locally, run `composer install` to get all dependencies, Run `npm run dev` and serve with `php artisan serve`
2. When everything looks fine, run `npm run prod`.
3. Copy the whole project folder to the deployment server (i.e., what's left with the .gitignore)
   a. If this is the first time, or a dependency has changed, run composer install/update
   b. else continue
4. Website is live.

Right so far?

Now, if someone now changes content in the production environment via control panel, /content will change, and it would be best for me to sync these changes back to my local environment - is that correct?

Thanks in advance!

**Comment by calebdw** [comment]
A lot of developers prefer to automate deployment as part of the CI/CD pipeline (after any tests pass of course). Check out [PHP Deployer](https://deployer.org), it's fairly straight-forward and can use in a pipeline or 'manually' via the terminal.

If you do choose to include `/content` in your repo then you can configure the production server to commit any content changes via `config/statamic/git.php`. If everything is configured correctly then you would just need to run a `git pull --rebase` on your local machine to sync the changes.

**Reply by FoxDawg** [reply]
Thanks for the reply. Seems like a good option, but looks like you would need root privileges on the hosting environment and npm installed.
But thanks for pointing me at the configuration, might give that a try as well.

**Reply by calebdw** [reply]
> looks like you would need root privileges on the hosting environment and npm installed.

If you're talking about PHP Deployer then not necessarily...you already have to have write permissions in the remote directory to copy files over. You can also run `npm` locally (or in the pipeline) and have Deployer copy the assets over---no need for `npm` on host machine.

**Reply by FoxDawg** [reply]
Will give it a try, thanks!
